projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -243,23 +314,38 @@ private void checkPermissionsThenCreatePdf() {
       }
       // Devices running SDK <= 22, permissions granted on installation
     } else {
-      createPdf();
+      permissionGranted = true;
+      createOutputDir();
     }
   }
 
-  // Call createPdf() if permission granted
   @Override
   public void onRequestPermissionsResult(
       int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
     if (requestCode == STORAGE_PERMISSION_CODE) {
       if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-        createPdf();
+        permissionGranted = true;
+        createOutputDir();
       } else {
```

Comment:
Something suppose to happen if someone decline? like returning to the main activity or closing the app

Reply:
At the moment it goes like this:
https://github.com/hasadna/hasadna/pull/115#discussion_r259969397